### PR TITLE
fix: ColorPicker UI in Dark Mode

### PR DESCRIPTION
**Pull request by RedJue**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution
Before：
image
After: 
image

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Fix the problem of abnormal style of color selector in dark mode       |
| 🇨🇳 Chinese |    修复在暗黑模式下颜色选择器的样式异常的问题      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 85a6607</samp>

Improved the color picker component's contrast and visibility in different color themes. Adjusted the color values of the presets based on the `theme` module and the `colorFillSecondary` token. Refactored and moved some style functions and removed unused code.

### <samp>🤖 Generated by Copilot at 85a6607</samp>

* Import `Color` type from `@rc-component/color-picker` as `RcColor` and `theme` module in `ColorPresets.tsx` ([link](https://github.com/ant-design/ant-design/pull/42827/files?diff=unified&w=0#diff-8d2a305db0c4945017603df77e97b2bf27dbde09aef2a8e2c2f7d10e67a3a822L1-R1),[link](https://github.com/ant-design/ant-design/pull/42827/files?diff=unified&w=0#diff-8d2a305db0c4945017603df77e97b2bf27dbde09aef2a8e2c2f7d10e67a3a822R8))
* Modify `isBright` function to take `bgColorToken` parameter and use `onBackground` method from `RcColor` to adjust color value based on background color in `ColorPresets.tsx` ([link](https://github.com/ant-design/ant-design/pull/42827/files?diff=unified&w=0#diff-8d2a305db0c4945017603df77e97b2bf27dbde09aef2a8e2c2f7d10e67a3a822L26-R32))
* Destructure `colorBgElevated` theme token from `useToken` hook and pass it to `isBright` function in `ColorPresets.tsx` ([link](https://github.com/ant-design/ant-design/pull/42827/files?diff=unified&w=0#diff-8d2a305db0c4945017603df77e97b2bf27dbde09aef2a8e2c2f7d10e67a3a822R39-R41))
* Update `isBright` function call to include `colorBgElevated` parameter in `ColorPresets.tsx` ([link](https://github.com/ant-design/ant-design/pull/42827/files?diff=unified&w=0#diff-8d2a305db0c4945017603df77e97b2bf27dbde09aef2a8e2c2f7d10e67a3a822L69-R78))
* Move `getTransBg` function from `color-block.ts` to `ColorPresets.tsx` and modify it to take `colorFill` parameter for the fill color of the transparent dots ([link](https://github.com/ant-design/ant-design/pull/42827/files?diff=unified&w=0#diff-c24d0ab998b998a6a5cfd4d77148aad78198e1fe93395501020c8bbee7ebfc1cL4-R8))
* Update `getTransBg` function call to include `colorFillSecondary` theme token as `colorFill` parameter in `color-block.ts` and `picker.ts` ([link](https://github.com/ant-design/ant-design/pull/42827/files?diff=unified&w=0#diff-c24d0ab998b998a6a5cfd4d77148aad78198e1fe93395501020c8bbee7ebfc1cL24-R22),[link](https://github.com/ant-design/ant-design/pull/42827/files?diff=unified&w=0#diff-34b89f88ff38f571cda106c0e4980f0487a530c9c21623a0fd1695bb0b132a58L61-R61))
* Adjust `insetInlineStart` property of the `&-color-checked::after` selector to align the check mark icon with the center of the color preset in `presets.ts` ([link](https://github.com/ant-design/ant-design/pull/42827/files?diff=unified&w=0#diff-258bf706e0141cb724422da77ea2a1f43ecad3cdaeee2f0b86908eb12f1af931L77-R76))
* Change `borderColor` property of the `&-color-bright::after` selector from `colorTextTertiary` to `rgba(0, 0, 0, 0.45)` to make the check mark icon more visible on bright colors in `presets.ts` ([link](https://github.com/ant-design/ant-design/pull/42827/files?diff=unified&w=0#diff-258bf706e0141cb724422da77ea2a1f43ecad3cdaeee2f0b86908eb12f1af931L99-R98))
* Remove unused `colorTextTertiary` theme token from `genPresetsStyle` function in `presets.ts` ([link](https://github.com/ant-design/ant-design/pull/42827/files?diff=unified&w=0#diff-258bf706e0141cb724422da77ea2a1f43ecad3cdaeee2f0b86908eb12f1af931L19))

### CODE REVIEW

**Comment by yoyo837** [comment]
这个东东有用例能覆盖吗?

**Comment by zombieJ** [comment]
不来个截图么~~

**Comment by RedJue** [comment]
> 不来个截图么~~

我加一个哈

**Review by kiner-tang — APPROVED** [review]

**Comment by RedJue** [comment]
> 这个东东有用例能覆盖吗?

好像木有 纯样式层面的

**Comment by yoyo837** [comment]
> > 这个东东有用例能覆盖吗?
> 
> 好像木有 纯样式层面的

取运行时的style或者getcomputedStyle判断呢

**Comment by yoyo837** [comment]
> > 这个东东有用例能覆盖吗?
> 
> 好像木有 纯样式层面的

深色模式下判断 `${colorPresetsPrefixCls}-color-bright` ?

**Comment by RedJue** [comment]
> cover

单测里怎么切换暗黑模式，有方法不?

**Comment by RedJue** [comment]
> 单测里怎么切换暗黑模式，有方法不?

File: components/color-picker/style/color-block.ts [hunk 1]
```diff
@@ -21,7 +19,7 @@ const genColorBlockStyle = (token: ColorPickerToken, size: number): CSSObject =>
       width: size,
       height: size,
       boxShadow: colorPickerInsetShadow,
-      ...getTransBg('50%'),
+      ...getTransBg('50%', token.colorFillSecondary),
```
Comment (zombieJ) [inline components/color-picker/style/color-block.ts:22]: 这个 token 用的是对的对吧？ @MadCcc

**Comment by RedJue** [comment]
> > > 这个东东有用例能覆盖吗?
> > 
> > 
> > 好像木有 纯样式层面的
> 
> 新代码cover的场景下判断 `${colorPresetsPrefixCls}-color-bright` 是否正确存在?

单测里怎么切换暗黑模式，有方法不?

**Comment by MadCcc** [comment]
> 单测里怎么切换暗黑模式，有方法不?

```
<ConfigProvider theme={{ algorithm: theme.darkAlgorithm }}>{...}</ConfigProvider>
```

File: components/color-picker/style/color-block.ts (hunk 1 above)
Comment (MadCcc) [reply components/color-picker/style/color-block.ts:22]: 从 Fill 里取的没什么问题，有问题以后再调整，现在看着挺舒服的

**Pull request merged**